upstream bug: http://bugs.php.net/bug.php?id=46311
The current patch follows a suggestion from Siarhei Siamashka
Adapted for php 5.5
--- Zend/zend_execute.h.old	2013-01-01 13:49:04.587825704 +0100
+++ Zend/zend_execute.h	2013-01-01 13:53:28.762958578 +0100
@@ -293,7 +293,7 @@
  	void **end = p - (int)(zend_uintptr_t)*p;
 
 	while (p != end) {
-		zval *q = *(zval **)(--p);
+		zval *q = (zval *)*(--p);
 		*p = NULL;
 		i_zval_ptr_dtor(q ZEND_FILE_LINE_CC);
 	}
